src/seedu/addressbook/AddressBook.java

```diff
@@ -1012,12 +1012,12 @@ private static String extractEmailFromPersonString(String encoded) {
 
         // email is last arg, target is from prefix to end of string
         if (indexOfEmailPrefix > indexOfPhonePrefix) {
-            return removePrefixSign(encoded.substring(indexOfEmailPrefix, encoded.length()).trim(),
+            return removePrefix(encoded.substring(indexOfEmailPrefix, encoded.length()).trim(),
                     PERSON_DATA_PREFIX_EMAIL);
 
         // email is middle arg, target is from own prefix to next prefix
         } else {
-            return removePrefixSign(
+            return removePrefix(
                     encoded.substring(indexOfEmailPrefix, indexOfPhonePrefix).trim(),
                     PERSON_DATA_PREFIX_EMAIL);
```

Comment:
The indentation for this part is a little off. `PERSON_DATA_PREFIX_EMAIL` is passed into `encoded.substring(...)` and thus should not be at the same indentation level as `encoded.substring(...)`.

Can you think of a more readable implementation? Perhaps extracting away `encoded.substring(...)` into another string variable before passing into `removePrefix(...)`?